google-cloud-logging/src/main/java/com/google/cloud/logging/SourceLocation.java

```diff
@@ -0,0 +1,163 @@
+   */
+  public static final class Builder {
+
+    private String file;
+    private Long line;
+    private String function;
+
+    Builder() {
+    }
+
+    Builder(SourceLocation sourceLocation) {
+      this.file = sourceLocation.file;
+      this.line = sourceLocation.line;
+      this.function = sourceLocation.function;
+    }
+
+    /**
+     * Sets the source file name. Depending on the runtime environment, this might be a simple name
+     * or a fully-qualified name.
+     */
+    public Builder setFile(String file) {
+      this.file = file;
+      return this;
+    }
+
+    /**
+     * Sets the line within the source file. 1-based; 0 indicates no line number available.
+     */
+    public Builder setLine(Long line) {
+      this.line = line;
+      return this;
+    }
+
+    /**
+     * Sets the human-readable name of the function or method being invoked, with optional context
+     * such as the class or package name. This information may be used in contexts such as the logs
+     * viewer, where a file and line number are less meaningful. The format can vary by language.
+     */
+    public Builder setFunction(String function) {
+      this.function = function;
+      return this;
+    }
+
+    public SourceLocation build() {
+      return new SourceLocation(this);
+    }
+  }
+
+  SourceLocation(Builder builder) {
+    this.file = builder.file;
+    this.line = builder.line;
+    this.function = builder.function;
+  }
+
+  /**
+   * Returns the source file name. Depending on the runtime environment, this might be a simple name
+   * or a fully-qualified name.
+   */
+  public String getFile() {
+    return file;
+  }
+
+  /**
+   * Returns the line within the source file. 1-based; 0 indicates no line number available.
+   */
+  public Long getLine() {
+    return line;
+  }
+
+  /**
+   * Returns the human-readable name of the function or method being invoked, with optional context
+   * such as the class or package name. This information may be used in contexts such as the logs
+   * viewer, where a file and line number are less meaningful. The format can vary by language.
+   */
+  public String getFunction() {
+    return function;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (!(obj instanceof SourceLocation)) {
+      return false;
+    }
+    SourceLocation other = (SourceLocation) obj;
+    return Objects.equals(file, other.file)
+        && Objects.equals(line, other.line)
+        && Objects.equals(function, other.function);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(file, line, function);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("file", file)
+        .add("line", line)
+        .add("function", function)
+        .toString();
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  LogEntrySourceLocation toPb() {
```

Comment:
Done, I had omitted it because `LogEntry`'s `toPb` function did, but didn't really think beyond that :P

I've changed it to `public`, and the one in `LogEntry` as well.